test/Microsoft.Azure.Devices.Edge.Test.Common/linux/PackageManagement.cs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Test.Common.Linux
+{
+    using System;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Serilog;
+
+    public class PackageManagement
+    {
+        bool isInitialized = false;
+
+        string os;
+        string version;
+        string packageTool;
+        SupportedPackageExtension packageExtension;
+        string forceInstallConfigCmd;
+        string installCmd;
+        string iotedgeServices;
+        string uninstallCmd;
+
+        public enum SupportedPackageExtension
+        {
+            Deb,
+            Rpm
+        }
+
+        public PackageManagement()
+        {
+            this.os = null;
+            this.version = null;
+            this.packageTool = null;
+            this.packageExtension = 0;
+            this.forceInstallConfigCmd = null;
+            this.installCmd = null;
+            this.iotedgeServices = null;
+            this.uninstallCmd = null;
+            this.isInitialized = false;
+        }
+
+        async Task CheckInitializationAsync()
+        {
+            if (!this.isInitialized)
+            {
+                var tokenSource = new CancellationTokenSource();
```

Comment:
Don't create a new token. These new `*Async` methods should take a `CancellationToken` argument, and should use the token that's passed in. That way if the test setup is taking too long it will time out.

Reply:
With the lazy<> lsbTask, passing in a token can be tricky. The currently solution is to have a hard-coded timeout of 10 seconds. (for ARM32).

Reply:
See my latest comments. We *need* to honor the cancellation tokens created in the fixtures.

Reply:
Yup. I got that figured out now. Sorry about that.